### PR TITLE
Improve CSV importer for crypto.com

### LINKED ISSUE #3056: Crypto CSV from CDC fails to import
Hi,
I tried to import import the CDC file ...Crypto.. it failed.

The other two Fiat and Cards worked fine.

CDC Rotki 3

CDC Rotki 2
CDC Rotki

I tried to send Excel CSV but it says this file not supported.

App Version: 1.17.2
Frontend Version: 1.17.2
Platform: win32 x64 10.0.19042
Electron: 12.0.7

Operating system: 
Rotki version:

### LINKED ISSUE #3418: Crypto.com fail import
## Problem Definition

Importing crypto transactions from Crypto.com app gives the error `Invalid CSV format, missing required field: Crypto.com csv missing entry for 'credited'`.
Debit card transactions work fine, but not crypto transactions.

## Logs

[rotkehlchen.log](https://github.com/rotki/rotki/files/7110897/rotkehlchen.log)

### System Description

Operating system: Windows 10
Rotki version:
```
App Version: 1.20.1
Frontend Version: 1.20.1
Platform: win32 x64 10.0.19042
Electron: 13.1.4
```

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #3418
Closes #3056

### CODE REVIEW

File: rotkehlchen/data/importer.py [hunk 1]
```diff
@@ -240,6 +240,7 @@ def _consume_cryptocom_entry(self, csv_row: Dict[str, Any]) -> None:
             'referral_card_cashback',
             'card_cashback_reverted',
             'reimbursement',
+            'viban_purchase',
```
Comment (LefterisJP) [inline rotkehlchen/data/importer.py:243]: is this a typo? What is viban? I see it below too. So probably not a typo. What is it?

File: docs/changelog.rst [hunk 2]
```diff
@@ -4,6 +4,8 @@ Changelog
 
 * :bug:`3406` Sushiswap events will now be taken into account during the profit and loss report.
 * :bug:`3407` Users will now see correctly reported their amount of xSUSHI.
+* :bug:`3418` Users importing data from crypto.com will now have a better experience when dust is not credited.
+* :bug:`3056` Users will now be better notified when an error occurred while importing information from crypto.com on credit/debit events with special cases.
```
Comment (LefterisJP) [inline docs/changelog.rst:8]: you are talking about better experience and better notifications here. But is that what you are doing? Doesn't your change fix the bug, thus making import actually work when doing those things?

There is a big difference between "improving experience" and "making it work"

File: rotkehlchen/data/importer.py [hunk 3]
```diff
@@ -406,18 +412,41 @@ def _import_cryptocom_associated_entries(self, data: Any, tx_kind: str) -> None:
...
         for row in data:
+            # If we don't have the corresponding debited entry ignore them
+            # and warn the user
+            if (
+                expects_debited is True and
+                row['Transaction Kind'] != f'{tx_kind}_debited'
+            ):
+                self.db.msg_aggregator.add_warning(
+                    f'Error during cryptocom CSV import consumption. Found {tx_kind}_credited '
+                    f'but no amount debited after at date {row["Timestamp (UTC)"]}',
```
Comment (LefterisJP) [inline rotkehlchen/data/importer.py:426]:
```suggestion
                    f'but no amount debited afterwards at date {row["Timestamp (UTC)"]}',
```

File: rotkehlchen/data/importer.py [hunk 4]
```diff
@@ -406,18 +412,41 @@ def _import_cryptocom_associated_entries(self, data: Any, tx_kind: str) -> None:
...
+                )
+                # Pop the last debited event as it's invalid
+                multiple_rows.pop(credited_timestamp, None)
+                # reset expects_credited value
```
Comment (LefterisJP) [inline rotkehlchen/data/importer.py:430]: typo? `expects_debited` ? I would remove the comment. It's really too obvious

File: rotkehlchen/data/importer.py [hunk 5]
```diff
@@ -406,18 +412,41 @@ def _import_cryptocom_associated_entries(self, data: Any, tx_kind: str) -> None:
...
+            # and warn the user
+            if (
+                expects_debited is True and
+                row['Transaction Kind'] != f'{tx_kind}_debited'
+            ):
+                self.db.msg_aggregator.add_warning(
+                    f'Error during cryptocom CSV import consumption. Found {tx_kind}_credited '
+                    f'but no amount debited after at date {row["Timestamp (UTC)"]}',
+                )
+                # Pop the last debited event as it's invalid
```
Comment (LefterisJP) [inline rotkehlchen/data/importer.py:428]: Why is it invalid? Write it in the comment.

**Review by LefterisJP — COMMENTED** [review]
Some questions. Also ... did you test this works and fixes the problems the user reported? I guess you could verify by the provided CSV examples.

File: rotkehlchen/data/importer.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/data/importer.py:243]: I saw this kind of transaction at #3056 By a search on google seems to be purchases using a fiat wallet

File: docs/changelog.rst (hunk 2 above)
Comment (yabirgb) [reply docs/changelog.rst:8]: Without this fix the csv import didn't get to finish and no data was imported. Now what I do, in the case of data that we expect and is missing, we notify the user about it and continue with the import 

in #3418 there was an amount that is "burned" but as how we know crypto.com works we expect two events, so we were expecting two events and only getting one. In this case we ignore this row and notify the user

File: rotkehlchen/data/importer.py (hunk 5 above)
Comment (yabirgb) [reply rotkehlchen/data/importer.py:428]: We always assume to be at least one debited event and one credited event. If this doesn't happen our logic breaks. And this should always happens but the output from crypto.com does strange things sometimes

File: docs/changelog.rst (hunk 2 above)
Comment (yabirgb) [reply docs/changelog.rst:8]: Also another thing is that credited/debited events always go on pairs and the timestamp is the same (this is the logic that was first implemented and what happens most of the time) but reading the #3056 csv file I found entries that were off by one second and this breaks things (in +1000 lines only happened twice).

That's why I talk about improving experience

**Comment by yabirgb** [comment]
Yes, I've tested with a bigger CSV having the same issue and it works fine :)

File: docs/changelog.rst (hunk 2 above)
Comment (LefterisJP) [reply docs/changelog.rst:8]: > Without this fix the csv import didn't get to finish and no data was imported

So why not mention that CSV import for crypto.com is actually working again. That's the only thing you need to say.

File: rotkehlchen/data/importer.py (hunk 5 above)
Comment (LefterisJP) [reply rotkehlchen/data/importer.py:428]: As I said ... write it in a condensed way in the comment in the code. Nobody will remember what you wrote here in the PR.

**Comment by LefterisJP** [comment]
Rebased on bugfixes since there was a conflict

**Review by LefterisJP — APPROVED** [review]
LGTM. Same tests as in develop failed.

**Pull request merged**